x/stakeibc/keeper/hooks.go

```diff
@@ -184,46 +184,42 @@ func (k Keeper) UpdateRedemptionRates(ctx sdk.Context, depositRecords []recordst
 	}
 }
 
-// Determine the undelegated balance from the deposit records queued for staking
-func (k Keeper) GetUndelegatedBalance(hostZone types.HostZone, depositRecords []recordstypes.DepositRecord) sdk.Dec {
-	// filter to only the deposit records for the host zone with status DELEGATION_QUEUE
-	undelegatedDepositRecords := utils.FilterDepositRecords(depositRecords, func(record recordstypes.DepositRecord) (condition bool) {
-		return ((record.Status == recordstypes.DepositRecord_DELEGATION_QUEUE ||
-			record.Status == recordstypes.DepositRecord_DELEGATION_IN_PROGRESS) &&
-			record.HostZoneId == hostZone.ChainId)
-	})
-
-	// sum the amounts of the deposit records
+// Determine the deposit account balance, representing native tokens that have been deposited
+// from liquid stakes, but have not yet been transferred to the host
+func (k Keeper) GetDepositAccountBalance(chainId string, depositRecords []recordstypes.DepositRecord) sdk.Dec {
+	// sum on deposit records with status TRANSFER_QUEUE or TRANSFER_IN_PROGRESS
 	totalAmount := sdkmath.ZeroInt()
-	for _, depositRecord := range undelegatedDepositRecords {
-		totalAmount = totalAmount.Add(depositRecord.Amount)
+	for _, depositRecord := range depositRecords {
+		transferStatus := (depositRecord.Status == recordstypes.DepositRecord_TRANSFER_QUEUE ||
+			depositRecord.Status == recordstypes.DepositRecord_TRANSFER_IN_PROGRESS)
+
+		if depositRecord.HostZoneId == chainId && transferStatus {
+			totalAmount = totalAmount.Add(depositRecord.Amount)
+		}
 	}
 
 	return sdk.NewDecFromInt(totalAmount)
 }
 
-// Determine the deposit account balance, representing native tokens that have been deposited
-// from liquid stakes, but have not yet been transferred to the host
-func (k Keeper) GetDepositAccountBalance(hostZone types.HostZone, depositRecords []recordstypes.DepositRecord) sdk.Dec {
-	// filter to only the deposit records for the host zone with status DELEGATION
-	depositAccountRecords := utils.FilterDepositRecords(depositRecords, func(record recordstypes.DepositRecord) (condition bool) {
-		return (record.Status == recordstypes.DepositRecord_TRANSFER_QUEUE ||
-			record.Status == recordstypes.DepositRecord_TRANSFER_IN_PROGRESS) &&
-			record.HostZoneId == hostZone.ChainId
-	})
-
-	// sum the amounts of the deposit records
+// Determine the undelegated balance from the deposit records queued for staking
+func (k Keeper) GetUndelegatedBalance(chainId string, depositRecords []recordstypes.DepositRecord) sdk.Dec {
```

Comment:
I would call all of these methods which end in Balance --> Total instead.  It is both clearer that the function is about to do a sum and return a single number but also it avoids confustion around the Balanced/Unbalanced delegations terminology

Reply:
I don't really see how the Balance here (noun - similar to the balance of a bank account) could be confused with the Balanced adjective used when describing a delegation

I think GetUndelegatedTotal is a bit unclear as it doesn't describe what the total is.

Would GetTotalUndelegatedBalance clarify it?

Reply:
A balance can only be the amount in a single account, in contrast a total is summed across different accounts.  This function is summing across deposit records to find a total but that total sits in several different validators accounts not one single account balance correct?  Or is this total we compute literally all sitting in the single module account "DelegationAddress" on the host chain but controlled by Stride?  If that is the case then I agree Balance makes more sense here

Reply:
Yeah so it's the latter, it's the balance of tokens sitting in the delegation account. So if you did a balance query for the delegation account - you would get the same number

The reason the implementation is a bit deceiving is because we can't simply do a balance query without issuing an ICQ (since it's on the host chain), so instead we turn to our internal record keeping which tracks the single balance across multiple records, divided up across epochs